### ISSUE TITLE
Provide a Callback for terminal.execute Function

### ISSUE THREAD

**Issue by ZitRos** [intersystems-ru/webterminal#107]
This issue is a result of the discussion with @wcjphoenix #106.

It would be nice to have an optional callback in `terminal.execute` API function, `execute(command, options, callback)`. The callback will receive the output after command execution is done. Example:

```javascript
window.onTerminalInit((terminal) => {
    terminal.execute("write $namespace", (outputChunks) => { // by default, execute does not echo output onto the user's screen
        const result = outputChunks.join("");
        if (result !== "USER") {
            terminal.print("\r\n\nPlease, change your namespace to "USER" by executing \x1b[31mzn "USER"\x1b[0m\r\n");
        }
    });
});
```

**Comment by ZitRos** [comment]
Done by 19dea6154ddb745b7fd983e197f459a82d546986.

**Comment by wcjphoenix** [comment]
Fantastic, thanks!

**Comment by ZitRos** [comment]
Please let me know if it works as expected! :)

**Comment by wcjphoenix** [comment]
It works well as long as there is not an Error in the code that is being executed.  However when there is an error, it overrides the echo=false, and prints to terminal then reprompts the user which triggers the OnOutput function which runs the execute and then causes the error to happen again causing an infinite loop.  Is there a reason that errors need to be echoed all the time and if so is it possible to have an option to turn that off?

Thanks,
Walter Jaskowiak

**Comment by ZitRos** [comment]
Oh yes, I need to fix this. Expect this fix to come tomorrow! Until then, I
hope the error you are getting is not a big problem.

Have a good weekend!
Nikita

**Comment by wcjphoenix** [comment]
Not at all, I just threw a try catch block around it.

Have a nice weekend!

**Comment by ZitRos** [comment]
I've fixed the error by c914b9f776929e6ded01d1e1de6d69fcd4799a6f (v4.6.1). Now the error message should go to the `outputChunks` array.
